components/Image/my-map-test.tsx

```diff
@@ -147,59 +168,52 @@ export default function MyMapTest({ source }: { source: string }) {
         setIsDragging(true)
       }
     }
-    setIsFirstPress(false)
-    setCfg({ ...cfg, isFirstPress: true })
   }
 
   const handleDoubleClick = (e: any) => {
     const { canMouseX, canMouseY } = setCoordinates(e)
-    if (imgScale === 3) {
-      setImgScale(1)
+    let tempScale
+    if (imgScale > 3) {
+      tempScale = 1
     } else {
-      let scaleHeight = Math.floor(natHeight * (imgScale + 0.5))
-      let scaleWidth = Math.floor(natWidth * (imgScale + 0.5))
-      setImgScale(imgScale + 0.5)
-      setSCHeight(scaleHeight)
-      setSCWidth(scaleWidth)
+      tempScale = imgScale + 0.5
     }
+    setNewImageLimits(tempScale, canMouseY, canMouseY)
+    setImgScale(tempScale)
   }
 
   const handleMouseMove = (e: any) => {
     let scaling = isScaling
     let tempImgScale: number = 1
+    let tempLeft = 0
+    let tempTop = 0
     const { canMouseX, canMouseY } = setCoordinates(e)
 
+    let scalediff = 0.025
```

Comment:
Here is one of these numbers that can go into a constant's object above.

Reply:
Done